### ISSUE TITLE
unexpected behavior: files module cannot write to /usr/local/

### ISSUE THREAD

**Issue by michaelhaaf** [blue-build/cli#150]
#### Current Behavior

Using the files module, when writing to /usr/local/, the following build-error arises:

```
========== Start Files module ==========
Adding files to image
Copying /tmp/config/files/usr to /usr
cp: cannot overwrite non-directory '/usr/local' with directory '/tmp/config/files/usr/local'
Error: building at STEP "RUN --mount=type=tmpfs,target=/var --mount=type=bind,from=stage-config,src=/config,dst=/tmp/config,rw --mount=type=bind,from=stage-modules,src=/modules,dst=/tmp/modules,rw --mount=type=bind,from=ghcr.io/blue-build/cli:exports,src=/exports.sh,dst=/tmp/exports.sh --mount=type=cache,dst=/var/cache/rpm-ostree,id=rpm-ostree-cache-combined-nvidia-latest,sharing=locked echo "========== Start Files module =========="   && chmod +x /tmp/modules/files/files.sh   && source /tmp/exports.sh   && /tmp/modules/files/files.sh '{"type":"files","files":[{"usr":"/usr"}]}'   && echo "========== End Files module =========="   && ostree container commit": while running runtime: exit status 1
ERROR => Failed to build combined-nvidia:local-39
```

See:
- my ["control" and "bugged" build on GitHub Actions](https://github.com/michaelhaaf/bluebuild-custom/actions/runs/8493910501/job/23268475395)
- the [minimally modified recipe.yml](https://github.com/michaelhaaf/bluebuild-custom/commit/3d1f6087dae2f186baca4378fa6453120fe26ed9) I used. 
 
Originally I reproduced this locally on a `bluebuild-alpine` distrobox (environment information follows below).

#### Expected Behavior

The successful build shows that the file module is capable of writing to `/usr/share/` and `/usr/etc/`. My expectation is that `/usr/local/` would work the same.

Example use case: manually adding system fonts that are not NerdFonts or GoogleFonts using a script. To my understanding, /usr/local/ is the recommended location, [see Fonts#Manual_installation in ArchWiki](https://wiki.archlinux.org/title/Fonts#Manual_installation)

My expectation could be wrong, for e.g. it is known that `/etc` should not be written to and this is explained in the [BlueBuild](https://blue-build.org/reference/modules/files/) documentation and, I think, in the upstream ublueos & fedora documentation as well (at least, that's my memory... couldn't find an example actually).

#### Additional context/Screenshots

N/A

#### Possible Solution

Either a fix (not sure where to start) or a documentation update.

#### Environment
- Blue Build Version: 0.8.3
- Operating system: Alpine Linux 3.19.1
- Branch/Tag:  ()
- Git Commit Hash:

#### Shell
- Name: bash
- Version: GNU bash, version 5.2.21(1)-release (x86_64-alpine-linux-musl)
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
- Terminal emulator: <unknown terminal> <unknown version>

#### Rust
- Rust Version: rustc 1.76.0 (07dca489a 2024-02-04)
- Rust channel: 1.76.0-x86_64-unknown-linux-gnu release
- Build Time: 2024-03-29 18:26:48 +00:00

#### Recipe:

- [control](https://github.com/michaelhaaf/bluebuild-custom/blob/file-module-bug-demo/config/recipe-control.yml)
- [bugged](https://github.com/michaelhaaf/bluebuild-custom/blob/file-module-bug-demo/config/recipe-bugged.yml)

**Comment by fiftydinar** [comment]
`/usr/local` is a symlink to `/var/local.

Writing to `/var` in build-time is currently not supported. This is an issue which needs to be solved in upstream.

**Comment by fiftydinar** [comment]
As a workaround, you may try to use `tmpfiles.d`, as Universal Blue uses it for exact same purpose of copying files that you can't in build-time. You may take a look at their SeLinux fixes for some of their packages, which I can't link right now, as an inspiration.

**Comment by michaelhaaf** [comment]
Ohhh okay that does make sense. Thanks @fiftydinar. 

For other's reference here's [the upstream documentation about /var/ symlinks](https://docs.fedoraproject.org/en-US/fedora-silverblue/technical-information/#filesystem-layout) and the [some bluebuild documentation that discusses it too](https://blue-build.org/learn/mindset/#what-is-and-isnt-possible)

I feel a note about `/var/` and the common symlinks like `/usr/local/` would be useful in the [files module documentation](https://blue-build.org/reference/modules/files/) next to the note about `/etc/`. I don't mind writing it.

**Comment by fiftydinar** [comment]
> I feel a note about `/var/` and the common symlinks like `/usr/local/` would be useful in the [files module documentation](https://blue-build.org/reference/modules/files/) next to the note about `/etc/`. I don't mind writing it.

I agree, it would be nice to further document this.

**Comment by gmpinder** [comment]
Should this be moved to the website repo? I feel like that's where general documentation would be done

**Comment by gmpinder** [comment]
Yeah if you can update the website to point to the modules repo instead for files, I'll remove the docs from this repo. In which case I'll also move this ticket to modules since the docs would need to be updated there.
